types/types.go

```diff
@@ -260,7 +262,7 @@ type ManifestUpdateOptions struct {
 // only to make writing struct literals possible.
 type ManifestUpdateInformation struct {
 	Destination  ImageDestination // and yes, UpdatedManifest may write to Destination (see the schema2 → schema1 conversion logic in image/docker_schema2.go)
-	LayerInfos   []BlobInfo       // Complete BlobInfos (size+digest) which have been uploaded, in order (the root layer first, and then successive layered layers)
+	LayerInfos   []BlobInfo       // Complete BlobInfos (size+digest+urls+mediatype+annotations) which have been uploaded, in order (the root layer first, and then successive layered layers)
```

Comment:
This is not currently true; `copy.Image` sets `InformationOnly.LayerInfos` based on one of

- `Image.LayerInfos`, which is documented above as ”`MediaType` may be [only] optionally provided”, and says nothing about URLs/Annotations (in particular schema[12] `LayerInfos` does not set `MediaType`)
- The return value of `ReapplyBlob`/`PutBlob` (which is supposed to have “all data filled in” but actually all implementations just set digest + size)

AFAICS `LayerInfos` is currently used only in the schema1→schema2 conversion, which really only the size, so for now just documenting these two fields might be fine.